lua/nvim-lsp-ts-utils/inlay-hints.lua

```diff
@@ -2,99 +2,162 @@ local o = require("nvim-lsp-ts-utils.options")
 local u = require("nvim-lsp-ts-utils.utils")
 
 local api = vim.api
-
+local INLAY_HINTS_METHOD = "typescript/inlayHints"
 local M = {}
 
-M.state = {
-    enabled = false,
-    ns = api.nvim_create_namespace("ts-inlay-hints"),
-}
-
-function M.setup_autocommands()
-    u.buf_autocmd(
-        "TSLspImportOnCompletion",
-        "BufEnter,BufWinEnter,TabEnter,BufWritePost,TextChanged,TextChangedI",
-        "autocmd_fun()"
-    )
+local ns = api.nvim_create_namespace("ts-inlay-hints")
+M.ns = ns
+
+-- enabled[bufnr] = false
+M.enabled = {}
+
+local function resolve_bufnr(bufnr)
+    if bufnr == 0 then
+        bufnr = api.nvim_get_current_buf()
+    end
+    return bufnr
+end
+
+local function buf_enabled(bufnr)
+    bufnr = resolve_bufnr(bufnr)
+    if bufnr ~= nil then
+        return M.enabled[bufnr]
+    end
 end
 
-local function hide(bufnr)
-    api.nvim_buf_clear_namespace(bufnr, M.state.ns, 0, -1)
+local function set_buf_enabled(bufnr)
+    M.enabled[resolve_bufnr(bufnr)] = true
+end
+
+local function set_buf_disabled(bufnr)
+    M.enabled[resolve_bufnr(bufnr)] = false
```

Comment:
```suggestion
    M.enabled[resolve_bufnr(bufnr)] = nil
```

More of a housekeeping thing, but I think it's better to clear this completely for garbage collection.